### PR TITLE
DC-457: Render the messages sent by the customer advisors

### PR DESCRIPTION
**Pull request by anakos**
Defines the logic for rendering the response from customer advisors to the client

### CODE REVIEW

**Comment by diksta** [comment]
So, first thing I notice is it doesn't compile ?!? See travis build fail details

File: app/uk/gov/hmrc/securemessagerenderer/controllers/MessageRendererController.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  def render(id: String) = Action.async { implicit request =>
+    messageRepository.findAdvice(CustomerAdviceId(id)).map {
+     case Some(CustomerAdvice(_,subject,body,_)) =>
+       Ok(Html(body)).withHeaders(
+         "X-Title" -> UriEncoding.encodePathSegment(subject, "UTF-8")
+       )
+     case _ =>
+       Logger.warn(s"unknown message id specified in request $id")
+       NotFound(s"unknown message id $id")
```
Comment (diksta) [inline app/uk/gov/hmrc/securemessagerenderer/controllers/MessageRendererController.scala]: "message id" or should it be "customer advice id" ??

File: app/uk/gov/hmrc/securemessagerenderer/controllers/MessageRendererController.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import uk.gov.hmrc.securemessagerenderer.repository.MongoCustomerAdviceRepository
+
+trait MessageRendererController extends BaseController {
+  def messageRepository : CustomerAdviceRepository
+
+  def render(id: String) = Action.async { implicit request =>
+    messageRepository.findAdvice(CustomerAdviceId(id)).map {
+     case Some(CustomerAdvice(_,subject,body,_)) =>
+       Ok(Html(body)).withHeaders(
+         "X-Title" -> UriEncoding.encodePathSegment(subject, "UTF-8")
```
Comment (diksta) [inline app/uk/gov/hmrc/securemessagerenderer/controllers/MessageRendererController.scala]: I'm going to assume there is a good reason for the X-Title header

File: app/uk/gov/hmrc/securemessagerenderer/repository/MongoCustomerAdviceRepository.scala [hunk 3]
```diff
@@ -34,14 +35,24 @@ class MongoCustomerAdviceRepository(implicit mongo: () => DB)
   )
   with CustomerAdviceRepository {
 
-  override def createAdviceWith(body: String)(implicit ec: ExecutionContext): Future[CustomerAdvice] = {
-    val advicePersistenceModel = CustomerAdvicePersistenceModel(body)
+  override def createAdviceWith(subject: String, body: String)(implicit ec: ExecutionContext): Future[CustomerAdvice] = {
```
Comment (diksta) [inline app/uk/gov/hmrc/securemessagerenderer/repository/MongoCustomerAdviceRepository.scala]: I think there was some discussion on a previous story about the subject always being the same and therefore not needing to persist it, but it wasn't a story I worked on, so perhaps I'm wrong on that.

Does the subject change ??

File: conf/application.conf [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 # This is the main configuration file for the application.
 # ~~~~~
-
+include "common.conf"
```
Comment (diksta) [inline conf/application.conf]: There isn;t a common.conf here - is it coming from somewhere else? What's in it?

Comment (anakos) [inline conf/application.conf]: nope, you're right - this was missed in the last round of cleanup and should have been removed.

File: app/uk/gov/hmrc/securemessagerenderer/repository/MongoCustomerAdviceRepository.scala (hunk 3 above)
Comment (anakos) [inline app/uk/gov/hmrc/securemessagerenderer/repository/MongoCustomerAdviceRepository.scala]: You're correct - the subject doesn't change. Will remove from persistent storage.

**Comment by diksta** [comment]
Noticed when I compile this I see a warning:

[warn] /Users/rich/code/hmrc/secure-message-renderer/conf/app.routes:4: possible missing interpolator: detected interpolated identifier `$id`
[warn] GET        /message/:id     uk.gov.hmrc.securemessagerenderer.controllers.MessageRendererController.render(id: String)

Maybe we should fix?  Nor sure why the warning is there though :-(

**Pull request merged**